### PR TITLE
Creating new protocols

### PR DESCRIPTION
**Pull request by pranavrao1**

### CODE REVIEW

File: draft-abiggs-saag-key-management-service.md [hunk 1]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+    "credential": {
+      "bearer": "ZWU5NGE2YWYtMGE2NC0..."
+    }
+  }  
+  "method": "create",
+  "uri": "/cmk",
+  "assignedOrgId": "10992782-e096-4fd3-9458-24dca7a92fa5",
+  "keyUris": [
+    "kms://{hostname}/cmk/10992782-e096-4fd3-9458-24dca7a92fa5"
+  ]
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: client will not specific key in here.

File: draft-abiggs-saag-key-management-service.md [hunk 2]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+~~~
+JWE(K_ephemeral, {
+  "client": {
+    "clientId": "android_a6aa012a-0795-4fb4-bddb-f04abda9e34f",
+    "credential": {
+      "bearer": "ZWU5NGE2YWYtMGE2NC0..."
+    }
+  }  
+  "method": "create",
+  "uri": "/cmk",
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: this should be `kms://{hostname}/cmk/`

File: draft-abiggs-saag-key-management-service.md [hunk 3]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+~~~
+JWE(K_ephemeral, {
+  "client": {
+    "clientId": "android_a6aa012a-0795-4fb4-bddb-f04abda9e34f",
+    "credential": {
+      "bearer": "ZWU5NGE2YWYtMGE2NC0..."
+    }
+  }  
+  "method": "update",
+  "uri": "/cmk",
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: This should be `kms://{hostname}/cmk/<<key_id>>`

File: draft-abiggs-saag-key-management-service.md [hunk 4]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+  "state": "inactive"
+  "keyUris": [
+    "kms://{hostname}/cmk/10992782-e096-4fd3-9458-24dca7a92fa5"
+  ]
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: Single operation not bulk.
`"keyIds":"key_id"

File: draft-abiggs-saag-key-management-service.md [hunk 5]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+  response
+}
+~~~
+
+Response message example:
+
+~~~
+JWE(K_ephemeral, {
+  {
+    "uri": "kms://{hostname}/cmk/10992782-e096-4fd3-9458-24dca7a92fa5",
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: include status code here of `200`

File: draft-abiggs-saag-key-management-service.md [hunk 6]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+
+~~~
+JWE(K_ephemeral, {
+  "client": {
+    "clientId": "android_a6aa012a-0795-4fb4-bddb-f04abda9e34f",
+    "credential": {
+      "bearer": "ZWU5NGE2YWYtMGE2NC0..."
+    }
+  }  
+  "method": "get",
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: use "retrieve" instead of "get"

File: draft-abiggs-saag-key-management-service.md [hunk 7]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+  "uri": "/cmk",
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: This should be ``kms://{hostname}/cmk/<<key_id>>``

File: draft-abiggs-saag-key-management-service.md [hunk 8]
```diff
@@ -1766,6 +1766,166 @@ JWE(K_ephemeral, {
...
+
+Response message example:
+
+~~~
+JWE(K_ephemeral, {
+{
+  "status": 200,
+  "keyUris": [
+    "kms://{hostname}/cmk/10992782-e096-4fd3-9458-24dca7a92fa5"
+  ]
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: no bulk operation just keyURI

Comment (pranavrao1) [reply draft-abiggs-saag-key-management-service.md]: include state.

File: draft-abiggs-saag-key-management-service.md [hunk 9]
```diff
@@ -1766,6 +1766,160 @@ JWE(K_ephemeral, {
...
+JWE(K_ephemeral, {
+  "client": {
+    "clientId": "android_a6aa012a-0795-4fb4-bddb-f04abda9e34f",
+    "credential": {
+      "bearer": "ZWU5NGE2YWYtMGE2NC0..."
+    }
+  }  
+  "method": "create",
+  "uri": "kms://{hostname}/cmk/",
+  "assignedOrgId": "10992782-e096-4fd3-9458-24dca7a92fa5"
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: optional value

File: draft-abiggs-saag-key-management-service.md [hunk 10]
```diff
@@ -1766,6 +1766,160 @@ JWE(K_ephemeral, {
...
+~~~
+JWE(K_ephemeral, {
+  "client": {
+    "clientId": "android_a6aa012a-0795-4fb4-bddb-f04abda9e34f",
+    "credential": {
+      "bearer": "ZWU5NGE2YWYtMGE2NC0..."
+    }
+  }  
+  "method": "retrieve",
+  "uri": "/cmk",
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: kms://{hostname}/cmk

File: draft-abiggs-saag-key-management-service.md [hunk 11]
```diff
@@ -1766,6 +1766,160 @@ JWE(K_ephemeral, {
...
+  "org": "10992782-e096-4fd3-9458-24dca7a92fa5"
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: assignedorgid and optional

File: draft-abiggs-saag-key-management-service.md [hunk 12]
```diff
@@ -1766,6 +1766,160 @@ JWE(K_ephemeral, {
...
+Response message example:
+
+~~~
+JWE(K_ephemeral, {
+{
+  "status": 200,
+  "keys": [
+    {
+      "uri": "kms://{hostname}/cmk/10992782-e096-4fd3-9458-24dca7a92fa5",
+      "state": "active"
```
Comment (pranavrao1) [inline draft-abiggs-saag-key-management-service.md]: orgid if there exists one.

File: draft-abiggs-saag-key-management-service.md (hunk 10 above)
Comment (pranavrao1) [reply draft-abiggs-saag-key-management-service.md]: add specific URI as well.

**Review by balthorium — APPROVED** [review]

**Pull request merged**